aspnetcore/signalr/hubs.md

```diff
@@ -0,0 +1,98 @@
+
+[!code-csharp[Create and use hubs](hubs/sample/hubs/chathub.cs?range=10-14)]
+
+You can specify a return type and parameters, including complex types and arrays, as you would in any C# method. Data that you receive in parameters or return to the caller is communicated between the server and the client by using JSON. SignalR handles the binding of complex objects and arrays of objects automatically.
+
+## Connection IDs
+
+Each connection to a SignalR hub has a unique connection ID. This ID can be used by other parts of your application to send messages directly to a particular connection.
+
+In the context of your hub, you can determine current by referencing its `Context.ConnectionId` object.
+
+[!code-csharp[Connection Ids](hubs/sample/hubs/chathub.cs?range=20-24)]
+
+## The Clients object
+
+Each instance of the `Hub` class has a property named `Clients` that contains the following members for communication between server and client:
+
+| Property | Description |
+| ------ | ----------- |
+| All | Calls a method on all connected clients |
+| Caller | Calls a method on all connected clients |
+| Others | Calls a method on clients other than a specific connection |
+
+Additionally, the `Hub` class contains the following methods:
+
+| Method | Description |
+| ------ | ----------- |
+| AllExcept | Calls a method on all connected clients except a specified connection |
+| Client | Calls a method on a specific connected client |
+| Clients | Calls a method on specific connected clients |
+| Group | Sends a message to a group of connections  |
+| GroupExcept | Sends a message to a group of connections, excluding  the specified group |
+| Groups | Sends a message to multiple groups of connections  |
+| OthersInGroup | Sends a message to others in a group of connections  |
+| User | Sends a message to a specific user |
+| Users | Sends a message to multiple users |
+
+Each property or method in the preceding tables contains a `SendAsync` method. The `SendAsync` method allows you to supply the name and parameters of the client method to call.
+
+## Send messages to clients
+
+Yo make calls to specific clients, use the members of `Clients.Client` or `Clients.Clients`. In the following example, the `SendMessageToSingleConnection` method demonstrates sending a message to one specific connection, while the `SendMessageToMultipleConnections` method sends a message to the clients  stored in an array named `ids`.
+
+[!code-csharp[Send messages](hubs/sample/hubs/chathub.cs?range=15-24)]
+
+## Handle events for a connection
+
+The SignalR Hubs API provides the `OnConnectedAsync` and `OnDisconnectedAsync` events to manage and track connections. Use the `OnConnectedAsync` event to capture the connection ID of the incoming connection.
```

Comment:
> The SignalR Hubs API provides the `OnConnectedAsync` and `OnDisconnectedAsync` events to manage and track connections.

They aren't technically events and we should make sure to be clear here for users. Let's call them "virtual methods" (which is what they are)

> Use the `OnConnectedAsync` event to capture the connection ID of the incoming connection.

We don't want to make a big deal about Connection IDs, they're somewhat internal and we'd prefer people not use them unless necessary. Groups are the preferred mechanism for tracking connections. Perhaps "Override the `OnConnectedAsync` virtual method to perform actions when a client connects to the Hub, such as adding it to a group"